slangpy/tests/slangpy_tests/test_tensor.py

```
@@ -277,5 +278,54 @@ def test_primaltensor_sample_scalar(device_type: DeviceType):
     assert abs(float(result) - float(expected)) < 1e-5
 
 
+@pytest.mark.parametrize("device_type", helpers.DEFAULT_DEVICE_TYPES)
+def test_tensor_str(device_type: DeviceType):
+    """Tensor.__str__ should return the string representation of its numpy data."""
+    device = helpers.get_device(device_type)
+    data = np.array([1.0, 2.0, 3.0], dtype=np.float32)
+    tensor = Tensor.from_numpy(device, data)
+    s = str(tensor)
+    assert "1." in s and "3." in s
```

Ruff (0.16.0)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)